packages/webpack/src/ember-webpack.ts

```diff
@@ -427,20 +426,16 @@ const Webpack: PackagerConstructor<Options> = class Webpack implements Packager
           variantIndex,
           entrypointAssets.map(asset => 'assets/' + asset.name)
         );
-
-        for (let asset of entrypointAssets) {
-          nonLazyAssets.add(asset.name);
-        }
-      }
-      if (variant.runtime !== 'browser') {
-        // in the browser we don't need to worry about lazy assets (they will be
-        // handled automatically by webpack as needed), but in any other runtime
-        // we need the ability to preload them
-        output.lazyBundles = new Set();
-        for (let asset of assets) {
-          if (!nonLazyAssets.has(asset.name)) {
-            output.lazyBundles.add('assets/' + asset.name);
-          }
+        if (variant.runtime !== 'browser') {
+          // in the browser we don't need to worry about lazy assets (they will be
+          // handled automatically by webpack as needed), but in any other runtime
+          // we need the ability to preload them
+          output.lazyBundles.set(
+            id,
+            chunks
+              .filter(chunk => chunk.runtime?.includes(id) && !entrypointAssets?.find(a => a.name === chunk.files?.[0]))
```

Comment:
I can't find any obvious docs about `chunk.runtime` but it appears to mean what we need it to mean: which entrypoints can use this chunk at runtime?